### PR TITLE
classnames: export types

### PR DESCRIPTION
**Pull request by JoshMcCullough**
- [ x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

This PR shouldn't change how people are currently using the types, but it exports types which are important in some circumstances.

### CODE REVIEW

File: types/classnames/classnames-tests.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import classNames = require('classnames');
-import * as classNames2 from 'classnames';
+import classNames2 from 'classnames';
```
Comment (seansfkelley) [inline types/classnames/classnames-tests.ts]: Why did this need to change? Did the supported import style change?

Comment (JoshMcCullough) [reply types/classnames/classnames-tests.ts]: Was not intentional -- fixed.

File: types/classnames/index.d.ts [hunk 2]
```diff
@@ -6,18 +6,11 @@
 //                 Sean Kelley <https://github.com/seansfkelley>
 //                 Michal Adamczyk <https://github.com/mradamczyk>
 //                 Marvin Hagemeister <https://github.com/marvinhagemeister>
+//                 Josh McCullough <https://github.com/joshmccullough>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-type ClassValue = string | number | ClassDictionary | ClassArray | undefined | null | boolean;
```
Comment (seansfkelley) [inline types/classnames/index.d.ts]: Could you just export the names are defined here instead of putting them in a new file?

**Comment by JoshMcCullough** [comment]
I tried but it broke the tests. I can try again maybe I missed something.

**Comment by seansfkelley** [comment]
That's legitimately surprising, unless it introduces some kind of name clash?

**Comment by JoshMcCullough** [comment]
Whoops, meant to comment inline. Since the current type def does `export = classNames;` specifically, you cannot export other things. I didn't want to remove that line because it is necessary AFAIK and if it's removed everything breaks.

**Comment by seansfkelley** [comment]
Arrggg, right, that makes sense. I guess if you need the types then there's no other way around it.

¯\\\_(ツ)\_/¯

**Review by seansfkelley — APPROVED** [review]

**Pull request merged**